rust/pattern/variable/type_.rs

```diff
@@ -111,6 +118,9 @@ impl fmt::Display for TypeVariable {
             write!(f, "{}", self.label.as_ref().unwrap().label)?;
         }
         if self.is_type_constrained() {
+            if self.reference.is_visible() && self.label.is_some() {
+                f.write_str(",")?;
+            }
```

Comment:
Bit ugly, but this is the cleanest way to insert the missing comma in the `$x type some-type, value double;` case, e.g. 
`write_joined!` could possibly be tweaked to make this formatter nicer.